pkgs/build-support/node/import-npm-lock/default.nix

```diff
@@ -127,8 +127,54 @@ in
       cp "$packageLockPath" $out/package-lock.json
     '';
 
+  # Build node modules from package.json & package-lock.json
+  buildNodeModules =
+    let
+      cleanAttrs = lib.flip removeAttrs [ "npmRoot" "package" "packageLock" "nodejs" ];
+    in
+    { npmRoot ? null
+    , package ? importJSON (npmRoot + "/package.json")
+    , packageLock ? importJSON (npmRoot + "/package-lock.json")
+    , nodejs
+    , ...
```

Comment:
Based on #237216 this behavior of implicitly passing through derivation arguments seems not the best to me. Consider instead an explicit `derivationArgs` like `runCommandWith` performs.

Reply:
It's not the best but in this case I don't see how it can be avoided.